lightningd/log.c

```diff
@@ -557,6 +557,9 @@ void log_dump_to_file(int fd, const struct log_book *lr)
 	struct log_data data;
 	time_t start;
 
+	sprintf(buf, "%s\n", "Start of new crash log");
```

Comment:
Why not `strcpy(buf, "Start of new crash log\n");`?  Or even `write_all(fd, "Start of new crash log\n", strlen("Start of new crash log\n"));`, a technique used a few lines below?

Reply:
Thanks, I changed it to the later.

Reply:
Also added another optimization below where we unnecessarily calculate strlen